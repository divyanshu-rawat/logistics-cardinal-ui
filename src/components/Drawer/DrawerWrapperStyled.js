import styled from 'styled-components';

const DrawerStyled = styled('div')`
  position: fixed;
  top: 0;
  width: auto;
  height: 100%;
  outline: 0;
  z-index: ${({ theme }) => theme.zIndex.drawerContent};
  ${({ right }) => {
    if (right) {
      return `right: 0;`;
    }
    return `left: 0;`;
  }};
  > .react-reveal {
    position: relative;
    height: 100%;
  }
`;

export default DrawerStyled;
