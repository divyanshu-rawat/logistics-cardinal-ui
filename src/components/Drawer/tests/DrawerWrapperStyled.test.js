import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import DrawerWrapperStyled from '../DrawerWrapperStyled';
import Theme from '../../../themes/rooster';

const renderedComponent = (right = false) =>
  shallow(<DrawerWrapperStyled right={right} theme={Theme} />);

describe('<DrawerWrapperStyled />', () => {
  it('should render the component with default position', () => {
    const tree = toJson(renderedComponent());
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with the property right', () => {
    const tree = toJson(renderedComponent(true));
    expect(tree).toMatchSnapshot();
  });
});
