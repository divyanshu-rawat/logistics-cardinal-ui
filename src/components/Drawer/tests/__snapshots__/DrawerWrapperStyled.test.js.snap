// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerWrapperStyled /> should render the component with default position 1`] = `
<div
  className="DrawerWrapperStyled__DrawerStyled-jiHDjs bjfxRn"
/>
`;

exports[`<DrawerWrapperStyled /> should render the component with the property right 1`] = `
<div
  className="DrawerWrapperStyled__DrawerStyled-jiHDjs bveLKc"
/>
`;
